mudz.Common/Domain/Singleton.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace mudz.Common.Domain
+{
+	public class Singleton<T> where T : new()
```

Comment:
Switch this for a simpler pattern if you want. Logic through inheritance so you can chop and change singleton patterns fairly easily.